bundles/binding/org.openhab.binding.powermax/src/main/java/org/openhab/binding/powermax/internal/message/PowerMaxBaseMessage.java

```diff
@@ -0,0 +1,192 @@
+            receiveType = null;
+        }
+    }
+
+    /**
+     * Work to be done when receiving a message from the Visonic alarm system
+     *
+     * @return a new state containing all changes driven by the message
+     */
+    public PowerMaxState handleMessage() {
+        // Send an ACK if needed
+        if (isAckRequired()) {
+            PowerMaxCommDriver.getTheCommDriver().sendAck(this, (byte) 0x02);
+        }
+
+        logger.debug("{}message handled by class {}: {}", (receiveType == null) ? "Unsupported " : "",
+                this.getClass().getSimpleName(), this.toString());
+
+        return null;
+    }
+
+    /**
+     * @return the raw data of the message (buffer of bytes)
+     */
+    public byte[] getRawData() {
+        return rawData;
+    }
+
+    /**
+     * @return the identifying code of the message (second byte in the buffer)
+     */
+    public int getCode() {
+        return code;
+    }
+
+    /**
+     * @return the type of the message to be sent
+     */
+    public PowerMaxSendType getSendType() {
+        return sendType;
+    }
+
+    /**
+     * @return the type of the received message
+     */
+    public PowerMaxReceiveType getReceiveType() {
+        return receiveType;
+    }
+
+    /**
+     * @return true if the received message requires the sending of an ACK
+     */
+    public boolean isAckRequired() {
+        return receiveType == null || receiveType.isAckRequired();
+    }
+
+    @Override
+    public String toString() {
+        String str = "\n - Raw data = " + DatatypeConverter.printHexBinary(rawData);
+        str += "\n - type = " + String.format("%02X", code);
+        if (sendType != null) {
+            str += " ( " + sendType.getDescription() + " )";
+        } else if (receiveType != null) {
+            str += " ( " + receiveType.getDescription() + " )";
+        }
+
+        return str;
+    }
+
+    /**
+     * Instantiate a class for handling a received message The class depends on the message.
+     *
+     * @param message
+     *            the received message as a buffer of bytes
+     *
+     * @return a new class instance
+     */
+    public static PowerMaxBaseMessage getMessageObject(byte[] message) {
+
+        Class<?> cl;
+        try {
+            cl = PowerMaxReceiveType.fromCode(message[1]).getHandlerClass();
+        } catch (IllegalArgumentException e) {
+            cl = PowerMaxBaseMessage.class;
+        }
+        try {
+            return (PowerMaxBaseMessage) cl.getConstructor(byte[].class).newInstance(message);
+        } catch (InstantiationException e) {
+            return new PowerMaxBaseMessage(message);
+        } catch (IllegalAccessException e) {
+            return new PowerMaxBaseMessage(message);
+        } catch (NoSuchMethodException e) {
+            return new PowerMaxBaseMessage(message);
+        } catch (InvocationTargetException e) {
+            return new PowerMaxBaseMessage(message);
+        }
```

Comment:
My intention here is just to consider as a standard message any message that I don't handle in the binding.